include/switcher.php

```diff
@@ -55,22 +57,31 @@ protected function get_elements( $links, $args ) {
 			$locale = $language->get_locale( 'display' );
 			$classes = array( 'lang-item', 'lang-item-' . $id, 'lang-item-' . esc_attr( $slug ) );
 			$url = null; // Avoids potential notice
-
-			if ( $current_lang = $links->curlang->slug == $slug ) {
-				if ( $args['hide_current'] && ! ( $args['dropdown'] && ! $args['raw'] ) ) {
-					continue; // Hide current language except for dropdown
-				} else {
-					$classes[] = 'current-lang';
+			$current_lang = ''; // Avoids potential notice
+
+			if ( 0 === $args['admin_render'] ) {
+				if ( $current_lang = $links->curlang->slug == $slug ) {
+					if ( $args['hide_current'] && ! ( $args['dropdown'] && ! $args['raw'] ) ) {
+						continue; // Hide current language except for dropdown
+					} else {
+						$classes[] = 'current-lang';
+					}
 				}
-			}
 
-			if ( null !== $args['post_id'] && ( $tr_id = $links->model->post->get( $args['post_id'], $language ) ) && $links->model->post->current_user_can_read( $tr_id ) ) {
-				$url = get_permalink( $tr_id );
-			} elseif ( null === $args['post_id'] ) {
-				$url = $links->get_translation_url( $language );
+				if ( null !== $args['post_id'] && ( $tr_id = $links->model->post->get( $args['post_id'], $language ) ) && $links->model->post->current_user_can_read( $tr_id ) ) {
+					$url = get_permalink( $tr_id );
+				} elseif ( null === $args['post_id'] ) {
+					$url = $links->get_translation_url( $language );
+				}
+			} else {
+				if ( isset( $args['lang'] ) && $args['lang'] == $slug ) {
+					if ( $args['hide_current'] && ! ( $args['dropdown'] && ! $args['raw'] ) ) {
+						continue; // Hide current language except for dropdown
+					}
+				}
```

Comment:
This is very similar to the condition above. I believe we should rather filter the languages list to exclude the current language before [iterating on it](https://github.com/polylang/polylang/blob/f6fbb96d14808fd5b54040c124aa237f61f0d3e5/include/switcher.php#L53).

Reply:
I agree that we should simplify this.
1. You are using two parameters, one documented ,`admin_render`, the other undocumented, `lang`. I would use only one. Let's call it `current_lang`, a it is what your `lang` is. You impose the current language in the language switcher.
2. The we could do something like:
```php
$current_lang  = empty( $args['current_lang'] ) ? $links->curlang->slug : $args['current_lang'];
```
This should allow you not to duplicate the test.
3. Then we should have to solve the issue with
```php
elseif ( null === $args['post_id'] ) {
				$url = $links->get_translation_url( $language );
				}
```
This could be done by checking `$args['current_lang']` in the test to avoid the line to be executed.

Reply:
In some REST API requests (post save for example) the language we set in admin_current_language could be `null`.
So we should test something like that `if ( ! is_null( $args['admin_current_language'] && empty( $args['admin_current_lang'] ) )`. I think it is difficult to understand because a `null` value is considered as an `empty` value too.
Thus I prefered to use two parameters `admin_render` as before and the new `admin_current_lang` (replacing lang) for testing what you need to do.